apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java

```diff
@@ -373,7 +376,13 @@ public Blob getBlob(final String container, final String key) {
                if (attributes.contains(XATTR_CONTENT_MD5)) {
                   ByteBuffer buf = ByteBuffer.allocate(view.size(XATTR_CONTENT_MD5));
                   view.read(XATTR_CONTENT_MD5, buf);
-                  hashCode = HashCode.fromBytes(buf.array());
+                  byte [] etagBytes = buf.array();
+                  if (etagBytes.length == 16) {
+                     hashCode = HashCode.fromBytes(buf.array());
+                     eTag = String.format("\"%s\"", hashCode.toString());
+                  } else {
+                     eTag = new String(etagBytes);
```

Comment:
Should this have a second `Charset` parameter?

Reply:
Done